kserve/README.md

```diff
@@ -0,0 +1,121 @@
+# KServe
+
+KServe comes with two component:
+
+1. [KServe](#KServe)
+2. [KServe runtimes](#KServe-Runtimes)
+
+## KServe
+
+Contains deployment manifests for the KServe controller.
+
+- [kserve-controller](https://github.com/opendatahub-io/kserve)
+  - Forked upstream kserve/kserve repository
+
+## KServe runtimes
+
+Contains the runtime manifests for KServe.
+
+- [kserve-controller](https://github.com/opendatahub-io/kserve)
+  - Forked upstream kserve/kserve repository
+
+## Original manifests
+
+KServe also uses `kustomize` so we can directly use [their manifests](https://github.com/kserve/kserve/tree/master/config).
+
+* `default` is the entrypoint for CRDs, KServe controller and RBAC resources.
+* `runtimes` is the second entrypoint for the KServe runtimes. They are referenced separately, as these are not namespaced.
+
+### Updating the KServe manifests
+
+Use the script in `hack` to update the KServe manifests:
+
+```bash
+./hack/update-kserve-manifests.sh
+```
+
+## Overlays
+
+There are two overlays defined with the necessary changes for ODH:
+
+* [controller](./odh-overlays/controller)
+* [runtimes](./odh-overlays/runtimes)
+
+
+### Installation process
+
+Following are the steps to install Model Mesh as a part of OpenDataHub install:
+
+1. Install the OpenDataHub operator
+2. Make sure you install Service Mesh and Serverless components and configure them appropriately
+3. Create a KfDef that includes the KServe components and runtimes
+
+```
+apiVersion: kfdef.apps.kubeflow.org/v1
+kind: KfDef
+metadata:
+  name: opendatahub
+  namespace: opendatahub
+spec:
+  applications:
+    - kustomizeConfig:
+        repoRef:
+          name: manifests
+          path: odh-common
+      name: odh-common
+    - kustomizeConfig:
+        repoRef:
+          name: manifests
+          path: kserve
+      name: kserve
+  repos:
+    - name: manifests
+      uri: https://api.github.com/repos/opendatahub-io/odh-manifests/tarball/master
+  version: master
+```
+
+4. You can now create a new project and create an `InferenceService` CR.
+
+## Using KServe in ODH
+
+You can use the `InferenceService` examples from KServe. Make sure to include the additional annotation for OpenShift Service Mesh:
```

Comment:
This kserve manifest assumes that the cluster has istio or service-mesh installed as pre-requisite?

Reply:
Yes indirectly. KServe expects Knative to be installed, which depends on a networking layer which could either be Kourier or Istio. As we want to go with Istio (and KServe has that as default), this results in the expectation that both are installed. I think there is still some discussion on how the full setup process would be, right?

Reply:
Right, I just wanted to know what applications are required before installing these manifests, and I think it's good to have that listed somewhere.